### ISSUE TITLE
Avoid forcing "proxied URL" in case of transparent request

### ISSUE THREAD

**Issue by ykopel** [apache/trafficserver#1448]
In the proxy/http/HttpTransact.cc file there is a call to "set_url_target_from_host_field" on each request that doesn't contain the "target" hostname in its URL part.
This means that in transparent mode the client can send a request with "partial" URL like this:
```
GET /partial HTTP/1.1
Host: 192.168.0.100:8000
Accept-Encoding: identit
```
But the trafficserver will add the hostname even if its in transparent mode. Like this:
```
GET http://192.168.0.100:8000/partial HTTP/1.1
Host: 192.168.0.100:8000
Accept-Encoding: identit
```

I think It will be good to add to the code the check of the transparent status before deciding to add the hostname to the URL part.

Here is a diff for the HttpTransact.cc file:
```
diff --git a/trafficserver/proxy/http/HttpTransact.cc b/HttpTransact.cc
index 085ebb5..1d143c3 100644
--- a/trafficserver/proxy/http/HttpTransact.cc
+++ b/HttpTransact.cc
@@ -820,7 +820,7 @@ HttpTransact::EndRemapRequest(State *s)
       TRANSACT_RETURN(SM_ACTION_INTERNAL_CACHE_NOOP, NULL);
     }
 
-    if (!s->http_config_param->url_remap_required && !incoming_request->is_target_in_url()) {
+    if (!s->http_config_param->url_remap_required && !incoming_request->is_target_in_url() && !s->client_info.is_transparent) {
       s->hdr_info.client_request.set_url_target_from_host_field();
     }
 
@@ -7785,7 +7785,9 @@ HttpTransact::build_request(State *s, HTTPHdr *base_request, HTTPHdr *outgoing_r
     }
   }
 
-  if (s->current.server == &s->server_info && (s->next_hop_scheme == URL_WKSIDX_HTTP || s->next_hop_scheme == URL_WKSIDX_HTTPS ||
+  // If the connection is transparent and the target is in the URL - keep it there
+  if (s->current.server == &s->server_info && !(s->client_info.is_transparent && base_request->is_target_in_url()) &&
+                                              (s->next_hop_scheme == URL_WKSIDX_HTTP || s->next_hop_scheme == URL_WKSIDX_HTTPS ||
                                                s->next_hop_scheme == URL_WKSIDX_WS || s->next_hop_scheme == URL_WKSIDX_WSS)) {
     DebugTxn("http_trans", "[build_request] removing host name from url");
     HttpTransactHeaders::remove_host_name_from_url(outgoing_request);
```

**Comment by maskit** [comment]
@ykopel Could you send the patch as a pull request so we can run our CI build with your patch?
I'm not familiar with transparent mode but the motivation and fix sound reasonable to me.

**Comment by bryancall** [comment]
Please fix PR #1473 if you would like to get this issue resolved.
